### DISCUSSION TITLE
i get odd error on choco 2.2.2

### DISCUSSION THREAD

**Discussion by soheilamiri** [chocolatey/choco#3348] [Q&A] [answered]
i install choco 2.2.2 on my server windows core 2022. but every time i try to install a package i get error: 
===================================================
[192.168.250.11]: PS C:\> choco install netcat
Chocolatey v2.2.2
Installing the following packages:
netcat
By installing, you accept licenses for the packages.

Progress: Downloading netcat 1.12... 1%
Progress: Downloading netcat 1.12... 2%
Progress: Downloading netcat 1.12... 3%
Progress: Downloading netcat 1.12... 4%
Progress: Downloading netcat 1.12... 5%
Progress: Downloading netcat 1.12... 6%
Progress: Downloading netcat 1.12... 7%
Progress: Downloading netcat 1.12... 8%
Progress: Downloading netcat 1.12... 10%
Progress: Downloading netcat 1.12... 11%
Progress: Downloading netcat 1.12... 12%
Progress: Downloading netcat 1.12... 13%
Progress: Downloading netcat 1.12... 14%
Progress: Downloading netcat 1.12... 15%
Progress: Downloading netcat 1.12... 16%
Progress: Downloading netcat 1.12... 17%
Progress: Downloading netcat 1.12... 18%
Progress: Downloading netcat 1.12... 19%
Progress: Downloading netcat 1.12... 21%
Progress: Downloading netcat 1.12... 22%
Progress: Downloading netcat 1.12... 23%
Progress: Downloading netcat 1.12... 24%
Progress: Downloading netcat 1.12... 25%
Progress: Downloading netcat 1.12... 26%
Progress: Downloading netcat 1.12... 27%
Progress: Downloading netcat 1.12... 28%
Progress: Downloading netcat 1.12... 29%
Progress: Downloading netcat 1.12... 30%
Progress: Downloading netcat 1.12... 32%
Progress: Downloading netcat 1.12... 33%
Progress: Downloading netcat 1.12... 34%
Progress: Downloading netcat 1.12... 35%
Progress: Downloading netcat 1.12... 36%
Progress: Downloading netcat 1.12... 37%
Progress: Downloading netcat 1.12... 38%
Progress: Downloading netcat 1.12... 39%
Progress: Downloading netcat 1.12... 40%
Progress: Downloading netcat 1.12... 41%
Progress: Downloading netcat 1.12... 42%
Progress: Downloading netcat 1.12... 44%
Progress: Downloading netcat 1.12... 45%
Progress: Downloading netcat 1.12... 48%
Progress: Downloading netcat 1.12... 51%
Progress: Downloading netcat 1.12... 55%
Progress: Downloading netcat 1.12... 58%
Progress: Downloading netcat 1.12... 62%
Progress: Downloading netcat 1.12... 65%
Progress: Downloading netcat 1.12... 68%
Progress: Downloading netcat 1.12... 72%
Progress: Downloading netcat 1.12... 75%
Progress: Downloading netcat 1.12... 78%
Progress: Downloading netcat 1.12... 81%
Progress: Downloading netcat 1.12... 85%
Progress: Downloading netcat 1.12... 88%
Progress: Downloading netcat 1.12... 92%
Progress: Downloading netcat 1.12... 95%
Progress: Downloading netcat 1.12... 98%
Progress: Downloading netcat 1.12... 100%

netcat v1.12.0 [Approved]
netcat package files install completed. Performing other installation steps.
The package netcat wants to run 'chocolateyInstall.ps1'.
Note: If you don't run this script, the installation will fail.
Note: To confirm automatically next time, use '-y' or consider:
choco feature enable -n allowGlobalConfirmation
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint):
Timeout or your choice of '' is not a valid selection.
You must select an answer
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint):
Timeout or your choice of '' is not a valid selection.
You must select an answer
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint):
Timeout or your choice of '' is not a valid selection.
You must select an answer
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint):
Timeout or your choice of '' is not a valid selection.
You must select an answer
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint):
Timeout or your choice of '' is not a valid selection.
You must select an answer
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint):
Timeout or your choice of '' is not a valid selection.
You must select an answer
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint):
Timeout or your choice of '' is not a valid selection.
You must select an answer
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint):
Timeout or your choice of '' is not a valid selection.
You must select an answer
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint):
Timeout or your choice of '' is not a valid selection.
You must select an answer
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint):
Timeout or your choice of '' is not a valid selection.
You must select an answer
Do you want to run the script?([Y]es/[A]ll - yes to all/[N]o/[P]rint):
Timeout or your choice of '' is not a valid selection.
You must select an answer
netcat not installed. An error occurred during installation:
 Too many bad attempts. Stopping before application crash.

Chocolatey installed 0/0 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
The process cannot access the file 'C:\ProgramData\chocolatey\lib\netcat\.chocolateyPending' because it is being used by another process.
======================================
this is a Bug !!!

**Comment by corbob** [comment, marked as answer]
I believe the answer is in the output:
> Note: If you don't run this script, the installation will fail.
> Note: To confirm automatically next time, use '-y' or consider:
> choco feature enable -n allowGlobalConfirmation

From the information provided, it looks like you tried to install it without confirming that you wanted to run the script, and so it failed to install.

**Reply by soheilamiri** [reply]
thanks, installation success by adding '-y' to end of the command but download process still show line by line
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %
Progress: Downloading vim 9.0.2056... %

**Reply by gep13** [reply]
Have a look at the `--no-progress` option, https://docs.chocolatey.org/en-us/choco/commands/install#options-and-switches

**Reply by soheilamiri** [reply]
is it normal to show progress line by line or i am doing some things wrong !!!

**Reply by gep13** [reply]
If you run the command in a normal PowerShell session, you don't see it line by line, but rather as a single line that updates each time.

However, when you are using other types of terminals, or executing remotely, or other places, you see each line being output, rather than being updated in place.
